management_app/views/faculty.py

```diff
@@ -8,9 +8,16 @@
 @faculty.route('/')
 @login_required
 def index():
+  # TODO: Get data from database
   faculties = [
     { 'name': 'Professor Mark',  'email': 'mark@uci.edu', 'role': 'Tenured Faculty', 'required_point': 3.5, 'prev_balance': 0.1250, 'estimated_balance': 0.2500 },
     { 'name': 'Professor 2',  'email': 'mark1@uci.edu', 'role': 'Pre Tenured Faculty 1 st', 'required_point': 2.5, 'prev_balance': 0.1000, 'estimated_balance': 0.1250 },
     { 'name': 'Professor 3',  'email': 'mark3@uci.edu', 'role': 'PoT Tenured Faculty', 'required_point': 6.5, 'prev_balance': 1.4375, 'estimated_balance': 0.9375 }
   ]
   return render_template('faculty/index.html', faculties=faculties)
+
+@faculty.route('/download_file/<filename>')
```

Comment:
If we are following REST principles then we should name the endpoint after a resource instead of an action. We could name it `/templates/<filename>` or `/data-templates/<filename>`. Also I think the preferred convention is to use hyphens instead of underscores when naming endpoints.

Reply:
Agree. Updated in the latest commit.